cmd/gateway/azure/gateway-azure.go

```diff
@@ -885,17 +885,24 @@ func (a *azureObjects) CopyObject(ctx context.Context, srcBucket, srcObject, des
 	if srcOpts.CheckCopyPrecondFn != nil && srcOpts.CheckCopyPrecondFn(srcInfo, "") {
 		return minio.ObjectInfo{}, minio.PreConditionFailed{}
 	}
-	srcBlobURL := a.client.NewContainerURL(srcBucket).NewBlobURL(srcObject).URL()
+	srcBlob := a.client.NewContainerURL(srcBucket).NewBlobURL(srcObject)
+	srcBlobURL := srcBlob.URL()
+
+	srcProps, err := srcBlob.GetProperties(ctx, azblob.BlobAccessConditions{})
+	if err != nil {
+		return objInfo, azureToObjectError(err, srcBucket, srcObject)
+	}
 	destBlob := a.client.NewContainerURL(destBucket).NewBlobURL(destObject)
+
 	azureMeta, props, err := s3MetaToAzureProperties(ctx, srcInfo.UserDefined)
 	if err != nil {
 		return objInfo, azureToObjectError(err, srcBucket, srcObject)
 	}
+	props.ContentMD5 = srcProps.ContentMD5()
```

Comment:
why did you change the md5sum?

Reply:
otherwise ListObjects has issues for the dest object as the Azure ETag is used - @kannappanr  noticed this with aws-cli

Reply:
Actually I tried to copy a multipart file in AWS S3 and it seems the ETag of the dest object is different to the source object.

```
11:36 $ mc stat s3/v-testbucket/6mb
Name      : 6mb
Date      : 2020-05-08 11:28:36 CET
Size      : 6.0 MiB
ETag      : 1300691e19c44c68501fcf2ca875f46e-2
Type      : file
Metadata  :
  Content-Type: application/octet-stream

Name      : 6mb.copy
Date      : 2020-05-08 11:36:18 CET
Size      : 6.0 MiB
ETag      : beb0ad7c527fdd8d8d56386b8a29b56e
Type      : file
Metadata  :
  Content-Type: application/octet-stream
```

However it is weird because the ETag of the dest object is the md5 sum of the source object.

Reply:
Yes, in the case of Azure

```
 {
            "Key": "128m",
            "LastModified": "2020-05-08T15:41:19.000Z",
            "ETag": "\"0x8D7F36640F24553-1\"",
            "Size": 134217728,
            "StorageClass": "STANDARD",
            "Owner": {
                "DisplayName": "",
                "ID": "02d6176db174dc93cb1b899f7c6078f08654445fe8cf1b6ce98d8855f66bdbf4"
            }
        },
        {
            "Key": "128m.copy",
            "LastModified": "2020-05-08T15:42:19.000Z",
            "ETag": "\"0x8D7F366647230FC-1\"",
            "Size": 134217728,
            "StorageClass": "STANDARD",
            "Owner": {
                "DisplayName": "",
                "ID": "02d6176db174dc93cb1b899f7c6078f08654445fe8cf1b6ce98d8855f66bdbf4"
            }
        },

```
```
$ mc stat myazure/testbucket/128m
Name      : 128m
Date      : 2020-05-08 08:41:19 PDT 
Size      : 128 MiB 
ETag      : 190d5e1aa1b3b051102415baccfc57ea-1 
Type      : file 
Metadata  :
  Content-Type: application/octet-stream 

Name      : 128m.copy
Date      : 2020-05-08 08:42:19 PDT 
Size      : 128 MiB 
ETag      : 0x8D7F366647230FC-1 
Type      : file 
Metadata  :
  Content-Type: application/octet-stream 
```
In the case of the original putobject, irrespective of this PR, there is mismatch between `etag` listed by azure and what gateway displays as part of `head-object`